src/message_buffer.cpp

```diff
@@ -80,128 +87,192 @@ MessageBuffer::operator=(MessageBuffer&& other)
   _buffer = std::move(other._buffer);
 }
 
+#ifndef htonll
+#define htonll(x) ((static_cast<uint64_t>(htonl(x)) << 32) + htonl((x) >> 32))
+#endif
+uint16_t
+hton(uint16_t value)
+{
+  return htons(value);
+}
+uint32_t
+hton(uint32_t value)
+{
+  return htonl(value);
+}
+uint64_t
+hton(uint64_t value)
+{
+  if constexpr (std::endian::native == std::endian::big)
+    return value;
+  else
+    return htonll(value);
+}
+
+template<typename Uint_t>
+MessageBuffer&
+operator<<(MessageBuffer& msg, Uint_t val)
+{
+  val = hton(val);
+  uint8_t* val_ptr = reinterpret_cast<uint8_t*>(&val);
+  msg._buffer.insert(msg._buffer.end(), val_ptr, val_ptr + sizeof(Uint_t));
+  return msg;
+}
+template MessageBuffer&
+operator<<(MessageBuffer& msg, uint16_t val);
+template MessageBuffer&
+operator<<(MessageBuffer& msg, uint32_t val);
+template MessageBuffer&
+operator<<(MessageBuffer& msg, uint64_t val);
+
+template<>
 MessageBuffer&
-operator<<(MessageBuffer& msg, const uint8_t val)
+operator<<(MessageBuffer& msg, uint8_t val)
 {
-  msg.push_back(val);
+  msg.push(val);
   return msg;
 }
 
+#ifndef ntohll
+#define ntohll(x) ((static_cast<uint64_t>(htonl(x)) << 32) + htonl((x) >> 32))
```

Comment:
Note it's the same definition as before.  So, we really only need a "swap byte order" defined once.  These could aliases for that.

Reply:
Very good point.